### ISSUE TITLE
Can't train MNIST digits to Perceptron

### ISSUE THREAD

**Issue by cazala** [wagenaartje/neataptic#27]
Hello, I'm having trouble to teach the MNIST dataset to a Perceptron. Here is a fiddle to see the code: https://jsfiddle.net/cazala/pn5zerq9/

I wanted to compare the performance to Synaptic's, since the readme claims to be 10x faster, and as the author of that library I'm very intrigued to see how you managed to get such a great speedup, but I couldn't make it train at all with a real dataset.

Here's that same piece of code using Synaptic so you get an idea of what I was trying to do: https://jsfiddle.net/cazala/672c4j1g/

I went over the docs but I couldn't find what am I doing wrong...

Oh, btw there seems to be a bug in the `schedule.function`, the docs say it receives a `data` object but I was getting `undefined`.. that's why line 15 is commented out.

**Comment by wagenaartje** [comment]
Hi. I just fixed the `data` bug. And while testing the MNIST dataset I realised that weights were initialised way too high for perceptrons. I fixed the JSFiddle (unfortunately, Neataptic has no async training). And I changed yours to output the time (and made it not async):


* Neatatpic: https://jsfiddle.net/pn5zerq9/3/
* Synaptic: https://jsfiddle.net/672c4j1g/2/

Results for my PC (chrome + windows 10.0.0):
* Neataptic: time = ~3100ms, error = 0.007
* Synaptic: time= ~29500ms, error = 0.009

I just did a few runs. Let me know if your results are different.
